File: Bubble Sort/Bubble Sort.cpp
// BUBBLE SORT IN C++

#include <iostream>
using namespace std;

#define length 10 // constant to set array size // constante para definir o tamanho do array

// function algorithm that returns the sorted array
// algoritmo da função que retorna o array ordenado
int * bubbleSort(int arr[])
{
	for (int i = 0; i < length - 1; i++)
	{
		for (int j = i + 1; j < length; j++)
		{
			if (arr[i] > arr[j])
			{
				int swap = arr[i];
				arr[i] = arr[j];
				arr[j] = swap;
			}
		}
	}

	return arr;
}

// main function only to call and test the sort function
// função principal apenas para chamar e testar a função de ordenação
int main(void)
{
	int arr[length];
	int *arrSorted;

	for (int i = 0; i < length; i++)
		cin >> arr[i];

	arrSorted = bubbleSort(arr);

	cout << endl << "Array Sorted:" << endl;

	for (int i = 0; i < length; i++)
		cout << arrSorted[i] << endl;

	return 0;
}

File: Bubble Sort/Bubble Sort.rs
// BUBBLE SORT IN RUST

// function algorithm that returns the sorted array
// algoritmo da função que retorna o array ordenado
fn bubble_sort(mut _arr: Vec<i32>) -> Vec<i32>
{
	for i in 0.._arr.iter().count() - 2
	{
		for _j in i + 1.._arr.iter().count() - 1
		{
			if _arr[i] > _arr[_j]
			{
				let swap = _arr[i];
				_arr[i] = _arr[_j];
				_arr[_j] = swap;
			}
		}
	}

	return _arr;
}

// main function only to call and test the sort function
// função principal apenas para chamar e testar a função de ordenação
fn main()
{
	let arr: Vec<i32> = vec![5, 2, -3, 10, 23, 99, -1, 7, 93, 0];
	let arr_sorted: Vec<i32> = bubble_sort(arr);
	
	println!("Array Sorted");

	for a in arr_sorted
	{
		println!("{}", a);
	}
}